packages/atomic/src/components/search/search-box-suggestions/suggestions-common.ts

```diff
@@ -3,41 +3,136 @@ import {VNode} from '@stencil/core';
 import {buildCustomEvent} from '../../../utils/event-utils';
 import {closest} from '../../../utils/utils';
 import {Bindings} from '../atomic-search-interface/atomic-search-interface';
+
+/**
+ * Element which will be rendered in the list of suggestions.
+ */
 export interface SearchBoxSuggestionElement {
+  /**
+   * Stable identity which enables Stencil to reuse DOM elements for better performance.
+   * The best way to pick a key is to use a string that uniquely identifies that list item among its siblings (often your data will already have IDs).
+   */
   key: string;
+  /**
+   * Rendered content of the element.
+   */
   content: Element | VNode;
+  /**
+   * Hook called when the selection is selected.
+   * @param e DOM event.
+   */
   onSelect?(e: Event): void;
-
+  /**
+   * The query associated with the suggestion, which will become the suggested query that would be sent if the search is executed.
+   */
   query?: string;
+  /**
+   * For improved accessibility, provide this property with additional information.
+   * https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label
+   */
   ariaLabel?: string;
+  /**
+   * Adds a specific shadow part attribute that can be selected with the CSS ::part pseudo-element.
+   * https://developer.mozilla.org/en-US/docs/Web/CSS/::part
+   */
   part?: string;
+  /**
+   * Hide the suggestion if it's the last in the list.
+   */
   hideIfLast?: boolean;
 }
+
+/**
+ * List of suggestions that will be displayed along other lists (e.g recent queries) when the search box's input is selected.
+ */
 export interface SearchBoxSuggestions {
+  /**
+   * The position of the list of the suggestions relative to the others.
```

Comment:
I think we should give some examples, clarifying that other suggestions default to their index in the list, and clarifying that the search box will simply sort suggestions by `position`.

Reply:
I am not sure I understand what this means :(

Reply:
Say you have three elements that each render a bunch of suggestions. Say they are initialized in the following order with the following positions:
* frequentlyAddedToCartSuggestions (position 1337)
* veryIntelligentSuggestions (position 20)
* recentQueriesSuggestions (position 1337)

They will be displayed under the search box in the following order:
* veryIntelligentSuggestions
* frequentlyAddedToCartSuggestions
* recentQueriesSuggestions

By default, when somebody uses our existing suggestion elements (e.g., `atomic-search-box-recent-queries`), the first element will choose `0` as the position for the suggestions it renders, the second will take `1`, the third will take `2`, and so on.

Reply:
Thank you!